src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/db/repositories/ReportRepository.kt

```diff
@@ -415,6 +417,21 @@ class OrderlyReportRepository(val isReviewer: Boolean,
                 .asTemporaryTable(name = "latest_version_for_each_report")
     }
 
+    override fun getLatestReportVersions(reports: List<String>): List<ReportWithDate>
+    {
+        JooqContext().use {
+            return it.dsl.select(
+                ORDERLYWEB_REPORT_VERSION_FULL.REPORT.`as`("name"),
+                ORDERLYWEB_REPORT_VERSION_FULL.DATE.max().`as`("date")
+            )
+                .from(ORDERLYWEB_REPORT_VERSION_FULL)
+                .where(shouldIncludeReportVersion)
```

Comment:
I think we don't need this `shouldIncludeReportVersion`, since that's actually to do with read permissions, whereas this is being called from an endpoint/controller method which requires run permission, and should be getting runnable reports.

(In practice, though,  it's unlikely a user would not be able to view a report that they could run. And there's also a wider issue here of where permissions are enforced - the repo needs to filter on permissions to get readable reports, but in general it's the endpoints that define and enforce permissions. Perhaps the methods in repo should at least be more clearly named to indicate their usage from the controllers. )

Reply:
Removed - thanks